### ISSUE TITLE
revealCursor moves the cursor to the first non-whitespace character

### ISSUE THREAD

**Issue by ygoe** [microsoft/vscode#132156]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.60.0, 1.59.1
- OS Version: Windows 10

Steps to Reproduce:

1. Set up the keybinding as described below
2. Open a document with indented lines of code
3. Move the cursor in the middle of a line of text, or at the very beginning (column 0)
4. Press Ctrl+Down until the cursor would move out of the viewport at the top (same for the other direction)

Now the cursor isn't where it was in its line anymore. If it was in the middle, now it's near the beginning. It seems to be pushed always to the first non-whitespace character. If the cursor was at the beginning of a line before scrolling, it will now move somewhere further right.

I'm used to keeping the cursor at the beginning of a line if possible. That makes line editing so much easier! But after scrolling, I have to clean up first. No other editor does that (Visual Studio, Notepad++ etc.).

This setting is often cited to achieve VS-like scrolling but that's not true. It has that quirks that gets in my way very often. Scrolling with VS Code ist just painful and probably the most bugging behaviour of this already strangely-behaving editor that feels like it was made for Mac exclusively (where every editor has its own keybindings, as opposed to Windows where most editors speak the same language).

Entry in keybindings.json: (among many more to make it really similar to Visual Studio and Notepad++)
```json
	{
		"key": "ctrl+up",
		"command": "editorScroll",
		"when": "editorTextFocus",
		"args": 
		{
			"to": "up",
			"by": "line",
			"revealCursor": true
		}
	},
	{
		"key": "ctrl+down",
		"command": "editorScroll",
		"when": "editorTextFocus",
		"args": 
		{
			"to": "down",
			"by": "line",
			"revealCursor": true
		}
	},
```

**Comment by rchiodo** [comment]
/verified AFAICT.

@ygoe can you try too? Thanks.

**Comment by ygoe** [comment]
Waiting for it to appear in a release build. I don't have experiments other than my own installed anywhere.

**Comment by ygoe** [comment]
Almost perfect in version 1.61: While the cursor remains at the same column as good as it can, it will move one line up if it was in the last empty line of a document, even before the cursor leaves the viewport. Not as annoying as before but still strange.

**Comment by ygoe** [comment]
I've just found a similar situation. I wanted Ctrl+PgUp/Down to behave as in Visual Studio. (*) I used this code, based on something I found, fixed to not scroll down a line for partially visible lines at the bottom.

```
	{
		"key": "ctrl+pageup",
		"command": "cursorMove",
		"args": {
			"to": "viewPortTop"
		},
		"when": "editorTextFocus"
	},
	{
		"key": "ctrl+pagedown",
		"command": "cursorMove",
		"args": {
			"to": "viewPortBottom",
			"value": 2
		},
		"when": "editorTextFocus"
	},
```

(The "value" parameter fixes the undesired scrolling.)

Again, the cursor is moved to the first non-whitespace character of the first or last visible line, respectively, no matter where it was before. There seems to be a pattern behind this. I just can't figure out what use that should have. Is that "first non-whitespace position" thing somehow intentional and we broke it now? Or is it a systematic error that might still appear in many other places?

*) Judging by the similarities in the keyboard hotkeys, VSCode should not have the name "Visual Studio" in it. There's almost no similarity, not even by Windows UI standards (Alt+Backspace isn't working by default).
